client/src/modules/list/components/Items/MoveToListPanel/MoveToListPanel.jsx

```diff
@@ -0,0 +1,207 @@
+import React, { Fragment, PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import _flowRight from 'lodash/flowRight';
+import { FormattedMessage, injectIntl } from 'react-intl';
+
+import Filter from 'common/components/Filter';
+import { fetchListsForItem } from 'modules/list/model/actions';
+import { moveItemToList } from 'modules/list/components/Items/model/actions';
+import { getListsForItem } from 'modules/list/model/selectors';
+import { IntlPropType, RouterMatchPropType } from 'common/constants/propTypes';
+import { CloseIcon } from 'assets/images/icons';
+import Confirmation from 'common/components/Confirmation';
+import { AbortPromiseException } from 'common/exceptions/AbortPromiseException';
+import { makeAbortablePromise } from 'common/utils/helpers';
+import Lists from './Lists';
+
+class MoveToListPanel extends PureComponent {
+  pendingPromise = null;
+
+  state = {
+    isConfirmationVisible: false,
+    listId: null,
+    listName: null,
+    listsToDisplay: [],
+    pending: false
+  };
+
+  componentDidMount() {
+    const { fetchListsForItem, lists } = this.props;
+
+    this.setState({ pending: true });
+
+    this.pendingPromise = makeAbortablePromise(fetchListsForItem());
+    this.pendingPromise.promise
+      .then(() => this.setState({ pending: false }))
+      .catch(err => {
+        if (!(err instanceof AbortPromiseException)) {
+          this.setState({ pending: false });
+        }
+      });
+
+    if (lists.length > 0) {
+      this.setState({ listsToDisplay: lists });
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const { lists: prevLists } = prevProps;
+    const { lists } = this.props;
+
+    if (prevLists.length === 0 && lists.length > 0) {
+      this.handleFilterLists(lists);
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.pendingPromise) {
+      this.pendingPromise.abort();
+    }
+  }
+
+  handleShowConfirmation = (listId, listName) => event => {
+    event.preventDefault();
+
+    this.setState({ isConfirmationVisible: true, listId, listName });
+  };
+
+  handleCloseConfirmation = () =>
+    this.setState({
+      isConfirmationVisible: false,
+      listId: null,
+      listName: null
+    });
+
+  handleClosePanel = event => {
+    const { onClose } = this.props;
+
+    this.handleCloseConfirmation;
+    onClose(event);
+  };
+
+  handleFilterLists = listsToDisplay => this.setState({ listsToDisplay });
+
+  handleMoveToList = event => {
+    event.preventDefault();
+
+    const { listId: newListId, listName } = this.state;
+    const {
+      data: { _id: itemId, name: itemName },
+      lockItem,
+      match: {
+        params: { id: listId }
+      },
+      moveItemToList,
+      unlockItem
+    } = this.props;
+    const notificationData = { item: itemName, list: listName };
+
+    lockItem();
+
+    return moveItemToList(itemId, listId, newListId, notificationData).finally(
```

Comment:
I would go for `moveItemToList(itemId, listId, newListId)(notificationData)`
where two-argument groups represent different context, first is related to moving item logic, second related to notification shown

Reply:
I can't be done like this, because this is `moveItemToList(itemId, listId, newListId)(dispatch)` because of thunk